tool_calling/tool_calling_llm.ipynb

```diff
@@ -0,0 +1,200 @@
+    "\n",
+    "## Prerequisites\n",
+    "\n",
+    "Before getting started, you'll need:\n",
+    "\n",
+    "- A [Writer AI Studio](https://app.writer.com/aistudio/signup?utm_campaign=devrel) account\n",
+    "- An API key, which you can obtain by following the [API Quickstart](https://dev.writer.com/api-guides/quickstart)\n",
+    "\n",
+    "## Setup\n",
+    "\n",
+    "Install the Writer SDK:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c05643760cecc348",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install -qU writer-sdk"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f3d9312e9a13f5e5",
+   "metadata": {},
+   "source": [
+    "Next, set the `WRITER_API_KEY` environment variable. We recommend setting it in a `.env` file in the root of your project, but this tutorial will set it in an environment variable if you don't have a `.env` file."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "785ea1629cdbb182",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import getpass\n",
+    "import os\n",
+    "from writerai import Writer\n",
+    "\n",
+    "if not os.getenv(\"WRITER_API_KEY\"):\n",
+    "    os.environ[\"WRITER_API_KEY\"] = getpass.getpass(\"Enter your Writer API key: \")\n",
+    "\n",
+    "client = Writer()  # The Writer client automatically uses the WRITER_API_KEY environment variable to authenticate."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8608bb9b23a5498e",
+   "metadata": {},
+   "source": [
+    "## Usage\n",
+    "\n",
+    "You've now initialized the Writer client and can use it to create a chat with an AI model.\n",
+    "\n",
+    "First, define the tools you want to provide to the LLM. In this example, you [define a tool](https://dev.writer.com/api-guides/model-delegation#tool-object) to use [Palmyra-Med](https://dev.writer.com/models/palmyra-med) to answer medical questions.\n",
+    "\n",
+    "The `type` field for the tool is `llm`. In the `function` field, the `model` is the ID of the [Palmyra model](https://dev.writer.com/home/models) you want to use. The `description` field describes when the tool should be used.\n",
+    "\n",
+    "> **Best practice:** The `description` field should indicate that the tool is an LLM. It should specify what the LLM is specialised in and when the tool should be used. The example below is a good description for a tool that invokes a medical LLM."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "83436249b52de650",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tools = [\n",
+    "    {\n",
+    "        \"type\": \"llm\",\n",
+    "        \"function\": {\n",
+    "            \"description\": \"A function that will invoke the llm identified by the given model, \"\n",
```

Comment:
maybe this is a dumb question, but why the line breaks here?

Reply:
They're not necessary, it was just for formatting b/c otherwise it scrolled pretty far off the cell.